File: Solutions/Solution359.java
class Logger {
  private Map<String, Integer> messageMap;

  public Logger() {
      this.messageMap = new HashMap<>();
  }
  
  public boolean shouldPrintMessage(int timestamp, String message) {
      int messageTimestamp = messageMap.getOrDefault(message, Integer.MIN_VALUE);

      if (messageTimestamp < timestamp - 9) {
          messageMap.put(message, timestamp);
          return true;
      } else {
          return false;
      }
  }
}

File: Solutions/Solution359.py
class Logger(object):

    def __init__(self):
        self.message_dict = {}

    def shouldPrintMessage(self, timestamp, message):
        """
        :type timestamp: int
        :type message: str
        :rtype: bool
        """
        if message not in self.message_dict or self.message_dict.get(message) < timestamp - 9:
            self.message_dict[message] = timestamp
            return True
        else:
            return False
        